### ISSUE TITLE
Install Certificate button broken?

### ISSUE THREAD

**Issue by casesolved-co-uk** [ddavness/power-mailinabox#55]
When I go to the `TLS (SSL) Certificates` page and try the `Install Certificate` button, nothing appears to happen except the page scrolls to the top?

Is this supposed to request and install a Let's Encrypt certificate?

I prefer doing each cert separately rather than the `Provision` button because there are domains in the list that are dealt with elsewhere on the server (I have a frontend nginx proxy outside the container).

Workaround:
From the power-maininabox git working copy run:
`management/ssl_certificates.py <domain name>`

Not sure if this was caused by the refactor?
https://github.com/ddavness/power-mailinabox/pull/41

**Comment by ddavness** [comment]
But it is working for me, though? I'll need more information to reproduce the issue.
_(To be able to get the button again - delete all files in `/home/user-data/ssl/` except the `/home/user-data/ssl/lets_encrypt/` folder, also delete all files in `/etc/nginx/conf.d/`, then re-run the setup accordingly)_

![image](https://user-images.githubusercontent.com/45321798/173677835-a4e4d013-37b2-4a63-8eeb-bf297f03f770.png)

> I prefer doing each cert separately rather than the `Provision` button because there are domains in the list that are dealt with elsewhere on the server (I have a frontend nginx proxy outside the container).

I recognize that as an niche-ier use-case (as the majority will definitely want to provision the certificates on all domains), but I'd probably accept a PR for it regardless. You can't go wrong with multiple ways to be able to do something important

**Comment by casesolved-co-uk** [comment]
So is the Install Certificate button linked with the external cert import form at the bottom?

**Comment by ddavness** [comment]
Ah... hold up - we were talking about different things.
The **Install Certificate** button allows you to import a certificate from somewhere else **manually**. It's not related to Let's Encrypt.

**Comment by casesolved-co-uk** [comment]
> I recognize that as an niche-ier use-case (as the majority will definitely want to provision the certificates on all domains), but I'd probably accept a PR for it regardless. You can't go wrong with multiple ways to be able to do something important. :)

Yeah I've just run into problems with this. I think it's a massive problem that if one host fails in a dns zone, the server refuses to provision anything.

It's a bit of a problem to fix though, because it tries to auto-provision anything that resolves to the server. It really needs an external request for each hostname.

I think a good fail-safe measure would be to do a second letsencrypt provision pass which excludes any failed hosts. What do you think?
